### ISSUE TITLE
Incorrect position pinning of Navigation Drawer

### ISSUE THREAD

**Issue by Aleksandyr** [IgniteUI/igniteui-angular#2013]
### Description  
When setting navdrawer position to right and pin it, nothing is changed and pinning happens on the left again.

 * igniteui-angular version: 6.1.x
 * browser: All

### Steps to reproduce  

1. Navigate to: https://staging.infragistics.local/products/ignite-ui-angular/angular/components/navdrawer.html#navigation-drawer-demo
2. Set position pinning to right
3. Pin the drawer.

### Result  
Drawer is pinned on the left.

### Expected result  
Drawer should be pinned on the right.

### Attachments  
![capture](https://user-images.githubusercontent.com/7037740/42757558-36af45a8-8909-11e8-890e-602d1179b4a1.PNG)

**Comment by damyanpetev** [comment]
Yup that's totally expected - see `position` in the [API](https://www.infragistics.com/products/ignite-ui-angular/angular/components/navdrawer.html#api). A pinned drawer becomes a part of the normal page layout and we can't directly control that in this component, so the position plays no role and that's why the radio group gets disabled.

Then again that was 2yrs ago and I'm totally down for changes.

**Comment by damyanpetev** [comment]
@Aleksandyr @SlavUI @kdinev Should we turn this into a feature request?

The nav darwer component can't control the entire app layout (that's kind of by design really) without a top-level `<app>`/`<lauoyt>` shell to integrate into, so we've documented the `pin` feature as application setup-dependent.

In a 5.x restyle we added a `flex-basis` width so the component can easily integrate inside applications using flex layout and it's possible to also add `order` that could enable the drawer to be pinned to the right. It will still work only in a flex layout (with some assumptions) and be completely dependent on the app to configure.

Thoughts?

**Comment by Aleksandyr** [comment]
@damyanpetev @SlavUI @kdinev In my opinion it would be nice if we could control the pinning of the nav-drawer.

If we are talking about mobile applications for instance it would be necessary to support that functionality because a lot of the applications are build with right pinned drawer which I think is pretty common scenario.
